pkg/workflows/upgrade.go

```diff
@@ -377,7 +409,7 @@ func (s *installCAPITask) Run(ctx context.Context, commandContext *task.CommandC
 	err := commandContext.ClusterManager.InstallCAPI(ctx, commandContext.ClusterSpec, commandContext.BootstrapCluster, commandContext.Provider)
 	if err != nil {
 		commandContext.SetError(err)
-		return &deleteBootstrapClusterTask{}
+		return &CollectMgmtClusterDiagnosticsTask{}
```

Comment:
Was this a bug fix or something?

Reply:
No, we did this because we no longer want to delete the bootstrap cluster if installing CAPI fails because we will now be able to resume the operation at this point